jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResource.java

```diff
@@ -208,7 +212,122 @@ public Response doResponseOk(final Entitlement createdEntitlement) {
         return callCompletionCreation.withSynchronization(callback, timeoutSec, callCompletion, callContext);
     }
 
+    @Timed
+    @POST
+    @Path("/createEntitlementWithAddOns")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Create an entitlement with addOn products")
+    @ApiResponses(value = {@ApiResponse(code = 400, message = "Invalid entitlement supplied")})
+    public Response createEntitlementWithAddOns(final List<SubscriptionJson> entitlements,
```

Comment:
@maguero See api comment i added in the [github issue](https://github.com/killbill/killbill/issues/199#issuecomment-153828997)